test/test.js

```diff
@@ -17,12 +24,37 @@ describe('primus-rooms-adapter', function () {
     expect(adapter.wildcard).to.be.an('object');
   });
 
-  describe('Adapter#del', function () {
+  describe.skip('Adapter#del', function () {
     it('should not leak memory', function () {
-      var adapter = new Adapter();
       adapter.del('foo');
       expect(adapter.rooms).to.eql({});
       expect(adapter.sids).to.eql({});
     });
   });
+  
+  describe('Adapter#add', function() {
+    it('should join a new room', function() {
+      let room = 'testRoom';
+      adapter.add(sparkID, room).then((res) => {
+        return adapter.get(sparkID);
+      }).then((res) => {
+        expect(res).to.be.instanceof(Array);
+        expect(res.length).to.eql(1);
+        expect(res[0]).to.eql(room);
+      });
+    });
+    
+    it('should join multiple rooms', function() {
+      const rooms = ['test1', 'test2', 'test3'];
+      let promises = [];
```

Comment:
Yes, my suggestion is to replace `Array.prototype.forEach()` with `Array.prototype.map()`.